File: server/bussines/reserva.js
const reservaDao = require("../dao/reservaDao.js");

const crearReserva = function(req, res){
    let reserva = 
    {
        fechaInicio: req.body.fechaInicio,
        fechaTermino: req.body.fechaTermino,
        total: parseInt(req.body.total),
        medioPago: req.body.medioPago,
        habitacionId: parseInt(req.body.habitacionId),
        usuarioId:  parseInt(req.body.usuarioId)        
    };
    reservaDao.crearReserva(reserva).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const actualizarReserva = function(req, res){
    let reserva = 
    {
        id: parseInt(req.body.id),
        fechaInicio: req.body.fechaInicio,
        fechaTermino: req.body.fechaTermino    
    };
    reservaDao.actualizarReserva(reserva).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};


const actualizarEstadoReserva= function(req, res){
    let reserva = 
    {
        id : parseInt(req.body.id),
        estado : parseInt(req.body.estado)
    };
    reservaDao.actualizarEstadoReserva(reserva).then((success) => {
        res.send(success.toString());
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerReservas = function(req, res){
    reservaDao.obtenerReservas().then((success) => {
        res.send(success);
    }).catch((error) => {
        console.log(error);
    });
};

const obtenerReservasPorUsuario = function ( req, res ) {
    reservaDao.obtenerReservasPorUsuario( req.body.rut ).then( ( success ) => {
        res.send( success );
    }).catch( ( error ) => {
        console.log( error );
    });
};

const obtenerReservasPorId = function ( req, res ) {
    reservaDao.obtenerReservasPorId( req.body.id ).then( ( success ) => {
        res.send( success );
    }).catch( ( error ) => {
        console.log( error );
    });
}

module.exports.crearReserva = crearReserva;
module.exports.actualizarReserva = actualizarReserva;
module.exports.actualizarEstadoReserva = actualizarEstadoReserva;
module.exports.obtenerReservas = obtenerReservas;
module.exports.obtenerReservasPorUsuario = obtenerReservasPorUsuario;
module.exports.obtenerReservasPorId = obtenerReservasPorId;

File: Desktop/Desktop.Data/Reserva.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desktop.Data
{
    public class Reserva
    {
        private string _url;
        public int ID { get; set; }
        public DateTime FechaInicio { get; set; }
        public DateTime FechaFin { get; set; }
        public int Total { get; set; }
        public int MedioPago { get; set; }
        public int Estado { get; set; }
        public int HabitacionID { get; set; }
        public Habitacion Habitacion { get; set; }
        public int UsuarioID { get; set; }
        public Usuario Usuario { get; set; }

        public Reserva()
        {
            _url = "http://127.0.0.1:3000/";
            GetHelper<Reserva>.Url = _url;
            Conexion.Cliente.BaseAddress = new Uri(_url);
        }

        public bool CrearReserva(Reserva reserva)
        {
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "", "" },
            };
            return Conexion.Post(queryParams, "crearReserva");
        }

        public bool ActualizarReserva(Reserva reserva)
        {
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "", "" },
            };
            return Conexion.Post(queryParams, "actualizarReserva");
        }

        public bool ActualizarEstadoReserva(int id, bool estado)
        {
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var queryParams = new Dictionary<string, string>
            {
                { "id", id.ToString() },
                { "estado", estado ? "1" : "0" },
            };
            return Conexion.Post(queryParams, "actualizarEstadoReserva");
        }

        public List<Reserva> ObtenerPromociones()
        {
            Conexion.Cliente.BaseAddress = new Uri(_url);
            var res = Conexion.Cliente.GetStringAsync(_url + "obtenerPromociones").Result;
            List<Reserva> reservas = JsonConvert.DeserializeObject<List<Reserva>>(res);
            return reservas;
        }
    }
}
